core/src/main/java/com/datastrato/gravitino/catalog/SchemaOperationDispatcher.java

```diff
@@ -297,18 +298,19 @@ public Schema alterSchema(NameIdentifier ident, SchemaChange... changes)
    */
   @Override
   public boolean dropSchema(NameIdentifier ident, boolean cascade) throws NonEmptySchemaException {
-    boolean dropped =
-        doWithCatalog(
-            getCatalogIdentifier(ident),
-            c -> c.doWithSchemaOps(s -> s.dropSchema(ident, cascade)),
-            NonEmptySchemaException.class);
-
-    if (!dropped) {
-      return false;
-    }
+    doWithCatalog(
+        getCatalogIdentifier(ident),
+        c -> c.doWithSchemaOps(s -> s.dropSchema(ident, cascade)),
+        NonEmptySchemaException.class);
 
+    // It could happen that the schema is not found in the catalog (dropped directly from
+    // underlying sources), but it is still in the store. So we should ignore the return value
+    // from catalog operation and try to delete the schema entity from the store.
     try {
       return store.delete(ident, SCHEMA, cascade);
+    } catch (NoSuchEntityException e) {
```

Comment:
Is the exception `NoSuchEntityException` thrown by the JDBC backend?

Reply:
Yes.